services/filemanager/fileManager_test.go

```diff
@@ -0,0 +1,339 @@
+package filemanager_test
+
+import (
+	"encoding/base64"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"syscall"
+	"testing"
+
+	"github.com/minio/minio-go/v6"
+	"github.com/ory/dockertest"
+	"github.com/rudderlabs/rudder-server/config"
+	"github.com/rudderlabs/rudder-server/services/filemanager"
+	"github.com/rudderlabs/rudder-server/utils/logger"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	AzuriteEndpoint, minioEndpoint string
+	base64Secret                   = base64.StdEncoding.EncodeToString([]byte(secretAccessKey))
+	bucket                         = "filemanager-test-1"
+	region                         = "us-east-1"
+	accessKeyId                    = "MYACCESSKEY"
+	secretAccessKey                = "MYSECRETKEY"
+
+	hold                bool
+	regexRequiredSuffix = regexp.MustCompile(".json.gz$")
+	fileList            []string
+)
+
+func TestMain(m *testing.M) {
+	config.Load()
+	logger.Init()
+
+	os.Exit(run(m))
+}
+
+//run minio server & store data in it.
+func run(m *testing.M) int {
+
+	flag.BoolVar(&hold, "hold", false, "hold environment clean-up after test execution until Ctrl+C is provided")
+	flag.Parse()
+
+	//docker pool setup
+	pool, err := dockertest.NewPool("")
+	if err != nil {
+		log.Fatalf("Could not connect to docker: %s", err)
```

Comment:
Lets avoid Fatal here:
```suggestion
		panic(fmt.Errorf("Could not connect to docker: %s", err))
```